pkg/operator/ceph/object/user/controller.go

```diff
@@ -493,7 +491,7 @@ func updateStatus(client client.Client, name types.NamespacedName, status string
 		return
 	}
 	if user.Status == nil {
-		user.Status = &cephv1.Status{}
+		user.Status = &cephv1.ObjectStoreUserStatus{}
 	}
 
 	user.Status.Phase = status
```

Comment:
I think you'll need to set `user.Status.SecretName` here (instead of line 348) so that every time `updateStatus()` is called, the secret name will be included in the update. Otherwise, the secret name will only be stored sometimes and will not be set other times. This should answer @sp98's concern as well.

Reply:
> I think you'll need to set `user.Status.SecretName` here (instead of line 348) so that every time `updateStatus()` is called, the secret name will be included in the update. Otherwise, the secret name will only be stored sometimes and will not be set other times. This should answer @sp98's concern as well.

Done.

Reply:
@subhamkrai IMO if u are setting `secret name` here, then there is chance in which `secret name` is refered even though user creation failed(and secret creation is also failed). Ideally `secret name` should be updated only if the secret is created. IMO keep creation of secret name in the ` generateCephUserSecret` itself and update it if status is `k8sutil.ReadyStatus`

Reply:
> @subhamkrai IMO if u are setting `secret name` here, then there is chance in which `secret name` is refered even though user creation failed(and secret creation is also failed). Ideally `secret name` should be updated only if the secret is created. IMO keep creation of secret name in the ` generateCephUserSecret` itself and update it if status is `k8sutil.ReadyStatus`

Done.